plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/notation/crowsfoot/CrowsFootDiagramNotation.java

```diff
@@ -115,6 +112,30 @@ private void createTargetDecorator(PolylineConnection conn, Color bckColor, Colo
         conn.setTargetDecoration(targetDecor);
     }
 
+    private List<DBSTableIndex> getIndexForSourceAttributes(
+        Collection<? extends DBSTableIndex> indexes,
+        List<ERDEntityAttribute> sourceAttr)
+        throws DBException {
+        List<DBSTableIndex> sourceAttributeIndexes = new ArrayList<>();
+        for (DBSTableIndex index : indexes) {
+            if (index.isPrimary()) {
+                // primary key index
+                continue;
+            }
+            VoidProgressMonitor monitor = new VoidProgressMonitor();
```

Comment:
Can we reuse the monitor as in the original code?

Reply:
+1. Pass monitor as parameter.

Generally I think that we need some global monmitor in Diagram or editor part. It will allow us to interrupt diagram layout (which may take a lot of time for huge diagrams). But we can create another ticket for that.

Reply:
Agreed, extracted monitor as argument now.

Reply:
Issue created: https://github.com/dbeaver/dbeaver/issues/22433